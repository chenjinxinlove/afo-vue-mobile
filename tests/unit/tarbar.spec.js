import {shallowMount, mount} from '@vue/test-utils'
import Demo from '../../src/tabbar/demo/'
import Vue from 'vue'
import TarBar from '../../src/tabbar/'
// import TarBarItem from '../../src/tabbar-item'

describe('Tarbar.vue', () => {
  // 测试是否可以被使用
  it('use', () => {
    Vue.use(TarBar)
    expect(Vue.component(TarBar.name))
      .toHaveProperty('name', 'VueComponent')
  })
  // 测试props
  it('props', () => {
    const wrapper = shallowMount(TarBar, {
      propsData: {
        title: 'title',
        fixed: true
      },
      slots: {
        default: '<TarBarItem>left</TarBarItem>'
      }
    })
    expect(wrapper.classes()).toEqual(['afo-tabbar', 'afo-tabbar--fixed'])
  })
  // demo测试快照
  it('demo snapshot', () => {
    expect(mount(Demo)).toMatchSnapshot()
  })
})
