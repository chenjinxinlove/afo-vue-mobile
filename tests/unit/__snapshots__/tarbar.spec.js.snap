// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tarbar.vue demo snapshot 1`] = `
<div>
  <div class="afo-tabbar afo-tabbar--fixed" style="z-index: 1; background: white;">
    <afo-tabbar-item>首页
    </afo-tabbar-item>
    <afo-tabbar-item icon="mall">购物车</afo-tabbar-item>
    <afo-tabbar-item icon="credit-card" dot="">订单</afo-tabbar-item>
    <afo-tabbar-item icon="person" info="12">个人中心</afo-tabbar-item>
  </div>
</div>
`;
